Title: Why Do Species Get a Thin Slice of π? Revisiting Lewontin's Paradox of Variation
Image: 
Summary: 
Status: draft

<meta name="twitter:card" content="summary" />
<meta name="twitter:site" content="@vsbuffalo" />
<meta name="twitter:title" content="Why Do Species Get a Thin Slice of π? Revisiting Lewontin's Paradox of Variation" />
<meta name="twitter:description" content="An overview of Buffalo (2021)." />
<meta name="twitter:image" content="http://vincebuffalo.org/images/buffalo_coop_fig1.png" />


The Great Obsession of population geneticists, to borrow John Gillespie's
words, is genetic variation. As an evolutionary biologist, it's rather hard to
*not* be obsessed with genetic variation, for it's the ultimate source of the
two most striking features of life on earth: the mind-boggling diversity of
species, and adaptations so utterly clever they look as though they were
assembled by a designer. Both life's dizzying diversity and cunning adaptations
are the result of evolutionary processes like natural selection and numerous
historical accidents, overlaid on one another like brush strokes on canvas, to
give us the snapshot of life on earth today.

Our obsession with genetic variation stems our view of evolution as the change
in the genetic composition of a population through time. By carefully studying
genetic variability, we hope we have some chance of figuring out *which*
evolutionary processes played out in the past. Some biologists dismiss this
view as "beanbag genetics", since population geneticists like to reduce a
population down to the simplest components: the frequencies of various genetic
variants, or alleles, in that population. While surely this is an
oversimplified view, the upside is that it is particularly amenable to thinking
about mathematically. Indeed evolution often occurs so slowly that to
understand it we need to use mathematics to figure out what a population looked
like long before we were born, or what it will look like long after we're dead.

Our field has constructed a rich mathematical theory of evolution over the last
one-hundred years, but it was only a half-century ago we got our first actual
estimates of the genetic variation in a fruitfly species named *Drosophila
pseudoobscura*, from the work of Richard Lewontin and Jack Hubby. After a
glimpse of the data, one theory of evolution was slain, and its rival theory
was in disarray. Within a few years, evolutionary biologists were measuring
genetic variation in all kinds of species during the "find 'em and grind 'em"
era, named for the unceremonious way numerous flies, crabs, plants, and
individuals from other species met their fate in the quest to measure
variability.

With new data of levels genetic variability *across* species, another theory
was soon on chopping block. This theory was the *neutral theory*, and Dick
Lewontin's 1974 synthesis of the find 'em and grind 'em era pointed out a
seeming contradiction between this theory and the new estimates of genetic
variability. He named this the "The Paradox of Variation", and it's an enduring
riddle in my field of evolutionary genetics. My recent paper published in eLife
attempts to make some progress on this longstanding paradox, and I'll give an
overview of the paper as well as some of the historical, personal, and
scientific context for this work.

## Lewontin and Hubby's Quest to Measure Genetic Variability

Before Lewontin and Hubby's work measuring genetic variability in *Drosophila
pseudoobscura*, evolutionary biologists were uncertain of how high or low the
genetic variability was within populations. As I write this, I am surrounded by
two examples of the extremes of this genetic variability spectrum: my *Monstera
deliciosa* houseplant (also known as the Swiss cheese plant for it's deeply
fenestrated leaves) and the *Drosophila* fruit flies that have invaded the
kitchen compost bin I've procrastinated emptying. While its incorrect to think
of the monsetera plants in people's homes as a "population" since they do not
interbred with one another, they serve as a good example a group of organisms
with low genetic diversity. This is because most monstera houseplants are
clones of one another, created by taking a clipping of one plant, letting it
root and grow, then taking a clipping of this plant, and so on. Unlike parent
and offspring which are separated by a generation, all monstera plants are
identical siblings; perhaps some mutations<sup></sup> lead to small amounts of
genetic variability, but this genetic variability is minuscule compared to
what's created by free mating in a sexually-reproducing species.

<figure>
<img src="/images/hubby_lewontin.png" width="460" class="img-responsive"/>
  <figcaption>
  </figcaption>
</figure>


By contrast, my compost bin population of *Drosophila melanogaster* harbors
massive amounts of genetic diversity. Part of the reason why *Drosophila
melanogaster* have such high genetic variability is because their population
sizes are so large. *Drosophila melanogaster* originated from equatorial
Africa, yet followed humans around the globe living off our trash heaps and
spoiled fruit. One way to measure genetic diversity is to take two random
chromosomes and count the differences between their DNA sequences, then repeat
this for another two random chromosomes, and so forth. Population geneticists
call this "pairwise" measure of genetic variability the Greek letter $\pi$ (I
imagine much to the disappoint of mathematicians). For *Drosophila*,
$\pi_\text{flies} \approx 1\%$, or one difference per 100 DNA basepairs, and
for humans, $\pi_\text{humans} \approx 0.1\%$, or one difference per 1,000
  basepairs.  Humans have much lower genetic diversity than fruitflies.
  Lewontin and Hubby explain the drastic importance of this simple quantity:

> In a sense, a description of the genetic variation in a population is the
> fundamental datum of evolutionary studies; and it is necessary to explain the
> origin and maintenance of this variation and to predict its evolutionary
> consequences. [...]

Thus, one end goal of measuring variability within and across species is to
answer the fundamental question of evolutionary genetics: what evolutionary
processes are compatible with the observed levels of genetic variation?

## The Evolutionary Theories Killed by Measuring Genetic Variation

<aside>
<sup></sup> This is how Lewontin himself sets up the problem in his 
book *The Genetic Basis of Evolutionary Change*.
</aside>

Before I introduce Lewontin's Paradox of Variation and the neutral theory, it's
worthwhile to set the stage with the theories of genetic variability that
preceded it<sup></sup>. At the time of Lewontin and Hubby's work, proponents of
the *classical theory* believed there wasn't going to be much genetic variation
between individuals.  This was because selection was thought to be extremely
powerful: a new mutation that increased survival or the number of offspring
quickly replaced all alternatives. The genetic composition of a population was
more or less uniform, as it moves from one perfectly adapted state to another
under the steady marching orders of natural selection. Of the variation that
existed, it was primarily from harmful mutations that 'broke' organisms from
their archetypal "wild-type"<sup></sup>. 
 
<aside>
If this view sounds almost vaguely political, you're in good company. Lewontin,
who never missed an opportunity to place scientific theories in
their historical context, writes:

"A basis for racism may also flow from the concept of a wild type, since if
there is genetic type of the species, those who fail to correspond to it must
be less perfect. Platonic notions of type are likely to intrude themselves from
one domain to another, and Dobzhanksy (1955) was clearly conscious of this
problem when he attacked the concept of a wild type." 
</aside>

By measuring their genetic variability, Lewontin and Hubby showed genetic
variability was far too abundant for the balance hypothesis to be correct. The
alternative view, the *balance theory*, held that natural selection maintained
vast amounts of genetic variation through a variety of mechanisms; aspects of
this theory were successfully challenged through other experiments and
arguments<sup></sup>. How evolutionary processes, whether random change,
natural selection, historical accidents (e.g. a population bottleneck), combine
to determine the central quantity of evolutionary genetics, genetic variation,
remained a mystery.

## The Neutral Theory

This brings us to the architect of the third controversial view of genetic
variation: Motoo Kimura. Kimura's view was that the majority of genetic
variation was *selectively neutral*: it did not impact the survival or number
of offspring individuals had. Free from the marching orders of selection, such
neutral variation lingered in populations, drifting up or down in frequency due
the vagaries of which variants are passed down to offspring (Mendelian
segregation), and random differences in survival and offspring number that
didn't have a genetic cause. We call this process *genetic drift*. Kimura's
neutral theory was in some ways an extension of Sewall Wright's view that
evolution was governed as much or more by random chance as it was by natural
selection. While here we're focused on Kimura's work on neutral polymorphism
within a population, much of Kimura's motivation for his neutral theory was to
explain observations in molecular evolution, such as the steady clock-like
ticking of amino acid differences between species. 

Suddenly under neutral theory, the mystery of why there was so much genetic
variation wasn't such a mystery. If the variation was neutral, natural
selection couldn't act on it. Surely, some beneficial mutations would enter the
population ---adaptations of course had a genetic basis--- but these mutations
were rare, and they would quickly replace their alternative and didn't persist
for long in the population. Lewontin called neutral theory the "neoclassical
  theory" because of it's similarity to the neoclassical theory.

Setting Goddess Tyche in charge of evolutionary changes has another benefit.
If we can ignore  natural selection, and thus all the uncertainty about the
*strength* and *direction* of natural selection, the mathematics of evolution
become rather simple. Under a simplified model where $N$ individuals freely
mate, new variation is created through mutation at rate $\mu$ per generation
per basepair (i.e. there's a $100 \times \mu$% chance that any one basepair
mutates in a given generation), and new variation is lost through random drift
at rate proportional to $1/N$ (i.e. slower in larger populations, fast in small
populations). Ultimately, like the level of water in a dam, a balance is met
between new mutations entering and old mutations going extinct in the
population: this is the equilibrium genetic diversity. A bit of mathematics
says that under this toy model, equilibrium genetic diversity should be $\pi
\approx 4N\mu$. In a big population (think of all the fruitflies in the world),
there should be high genetic variability. If population sizes are small (think
of a small island), soon everyone becomes everyone else's cousin, and genetic
diversity is lost.

## Lewontin's Paradox of Variation

So while neutral theory is mathematically alluring, and not at odds with the
high level of genetic variability found in various species, Lewontin noticed a
critical problem. Under Kimura's neutral theory, the heterozygosity at a locus
should depend on the product of mutation rate and population size, $N \mu$,
such that: 

$$
H = \frac{4N\mu}{1 + 4 N \mu}
$$

However, the range of genetic variability across species was surprisingly
narrow. Lewontin lays out why this is a problem visually in his book:

<figure>
<img src="/images/lewontin_1974.jpg" width="460" class="img-responsive"/>
  <figcaption>
  A figure from Lewontin's 1974 book *The Genetic Basis of Evolutionary Change*
  on the "Paradox of Variation" (p. 209). The curved line is the hypothetical
  heterozygosity (another term for genetic variability) given by $H = 4N\mu /
  (1 + 4N\mu)$. The range of variation observed in early studies ranges
  between $H \approx 6\%$ to $H \approx 20\%$, suggesting the product $N\mu$
  is between 0.015 and 0.06. For a gene mutation rate of $\mu = 10^{-6}$, this
  suggests that $N$ ranges from 3,750 to 15,000 --- a minuscule factor of
  four difference.  
  </figcaption>
</figure>

As Lewontin explains, 

> The observed range of heterozygosities over all the species [...] lies in the
> sensitive region, between 0.056 and 0.184. This range corresponds to values
> of $N\mu$ between 0.015 and 0.057. Since there is no reason to suppose that
> mutation rate has been specially adjusted in evolution to be the reciprocal
> of population size for higher organisms, we are required to believe that
> higher organisms including man, mouse, and Drosophila and the horseshoe crab
> all have population sizes within a factor of 4 of each other. [...] **The
> patent absurdity of such a proposition is strong evidence against a
> neutralist explanation of observed heterozygosity**. 

While Lewontin's estimated range of heterozygosities only included 13 species
(Table 22, p. 117), this problem wasn't just an artifact of small sample size.
Throughout the 1970s, estimating protein heterozygosities with gel
electrophoresis became a cottage industry. In the decade since Lewontin's book,
over a thousand species had published estimates of their protein variability,
which confirmed the narrow range of genetic variability found by Lewontin. Nevo
and colleagues published a survey of estimates of protein heterozygosities for
1,111 different species, finding that heterozygosity ranges only from 0% to
30%<sup></sup>:

<aside>
  There's a funny bit of personal history here as well. I attended many of the
  talks of a two-week Simons Institute workshop at UC Berkeley, but missed the
  first week, which was a sort of bootcamp for population geneticists. Later, I
  watched many of these talks including <a
  href="https://youtu.be/y0VjObP1lBA?t=1648">this lecture by Monty Slatkin</a>
  on the history of population genetics. Midway through, Monty shows the
  heterozygosity figure from Nevo et al. (1984) to introduce Lewontin's
  Paradox. This was the first time I'd heard of this problem. 
</aside>

<figure>
<img src="/images/nevo.png" width="460" class="img-responsive"/>
  <figcaption>
  Figure 2c from Nevo et al. *The Evolutionary Significance of Genetic
  Diversity: Ecological, Demographic and Life History Correlates* (1984).
  </figcaption>
</figure>

So Lewontin's Paradox of Variation seemed to be a rather serious problem for
Kimura's neutral theory, and soon folks were looking for other evolutionary
processes that were consistent with the observed data.

## Lewontin's Paradox and the Hitchhiking Model

Lewontin wrote his book while on sabbatical in John Maynard Smith's lab at the
University of Sussex. The neutral theory was under attack from multiple angles,
but especially so in Britain, where a cultural that treasured naturalism and
adaptive storytelling fostered the conviction that *every* genetic variant must
have *some* adaptive benefit<sup></sup>. As Maynard Smith described,

> The whole tradition of British population biology had
been, if you find a genetic variability, it must have some kind of selective
explanation, and if at first you don't find it, you must try, try, and try
again until you do.

Perhaps in the spirit of this tradition, Maynard Smith and John Haigh tried to
find a selective explanation for the observed narrow range of diversity
diversity. In 1974, they published their masterwork *The hitch-hiking effect of
a favourable gene*, which develops a mathematical model of how a selected
variant can decrease its surrounding genetic variability. When a new strongly
selected mutation occurs, it does so on a random chromosome in the population.
As this mutation increases in frequency, either by improving an individual's
odds of survival or number of offspring, it drags along its neighboring
stretches of DNA as it spreads within the population. The intuition behind why
is that our DNA is a bit like a spool of 8mm film, and you can consider our
genes as random frames of the motion picture. Most neighboring frames during a
scene of the film end up together into the final production, much like if I
pass on a variant I inherited from my father, there's a good chance that
variant's neighbor would also end up in my child. However, occasionally the
film editor comes along and cuts the scene short and splices in another scene.
So too can a cell's recombination machinery cut the stretch of my father's
chromosome I'm passing on and splice in my mother's; suddenly her genetic
variants and their neighbors go on to the next generation. Our genomes, after
all, are exactly half of each our parents' genomes, but a random mosaic of our
four grandparents' genomes.

<figure>
<img src="/images/film-splice.png" width="460" class="img-responsive"/>
  <figcaption>
  A Griswald Junior Film Splicer splicing together two bits of 16mm film (this still is from 
  <a href="https://www.youtube.com/watch?v=x7vrs67YDZI&t=281s">this short film</a>).
  </figcaption>
</figure>

This is *recombination*, and it's a vitally important part of the evolutionary
process. In particular, how often recombination occurs ---how often the scenes
of a film are cut and spliced to different scenes--- varies across organisms
and along the chromosome itself. In regions of high recombination, beneficial
variants are quickly spliced off and away from their neighbors, allowing the
surrounding genetic variation to persist in the population undisturbed. By
contrast, in low recombination regions, a beneficial variant won't be spliced
away from even its furthest neighbors for quite some time, and vast
neighborhoods of genetic diversity are wiped out whenever a beneficial mutation
comes along and takes over the population. It is precisely this process that
Maynard Smith and Haigh thought could be reducing genetic variability along the
genome, and explain the narrow range of genetic diversity seen across species.
As they say, 

> [This investigation] can therefore be regarded as a last ditch attempt to
> save the neutral mutation theory by showing that there is another process
> which can account for the uniformity of [heterozygosity] between species.

Nearly fifty years of population genetics has indeed shown that hitchhiking can
strongly impact genetic diversity within a species. The textbook example is
from work a member of my dissertation committee, Dave Begun, did as a PhD
student with Chip Aquadro. They showed that there was a striking correlation
between recombination rates and genetic diversity in *Drosophila melanogaster*,
which is precisely what one would expect under pervasive hitchhiking. Genetic
diversity is highest in high recombination regions, and lowest in low
recombination region across the *Drosophila melanogaster* genome:

<figure>
![](/images/begun_aquadro.png)
<figcaption>
The correlation between pairwise diversity (a measure of genetic variability)
and recombination rate <br/>in *Drosophila melanogaster*, from [Begun and
Aquadro (1992)](https://www.nature.com/articles/356519a0).
</figcaption>
</figure>

Later, theoretic work showed that a similar reduction in diversity can happen
if new mutations are *selected against*, rather than *selected for*. This
process is called *background selection*, as it's continually occurring in
the background of the genome. Population geneticists are still debating and
developing new ways to estimate the relative strengths of hitchhiking and
background selection, which collectively we call *linked selection*. While
hitchhiking and background selection can reduce genetic diversity, and have
been shown to do so in many species, the central question remained
unanswered: are these selection processes strong enough to constrain genetic
diversity across species to the narrow range observed? Or is there some other
explanation?

## The Neutralist View on Lewontin's Paradox

Kimura and others had a simpler explanation for the narrow range of diversity.
First, we have known since Sewall Wright that what is relevant to the rate of
genetic drift, and thus the level of genetic variability, isn't a species
population *census size* (i.e. the total number of individuals in the
population), but rather its *effective population size*.  Effective population
size is a bit of an amorphous concept in population genetics, but it's
fundamentally a way to account for the complexities of real populations of
breeding individuals. For many populations, the idealized random mating we
assume in our mathematical models is overly simplistic and ignores the
realities of ecology and demography. Fortunately, we have found that in the
vast majority of cases, we can account for the complexities of real populations
by simply rescaling the population size, $N$, to a new "effective population
size", $N_e$. 

Think of it this way: imagine 500 men and 500 women on an island. There's a
culture of free love and random mating. Still, it being an island, after a few
generations, everyone becomes everyone else's cousin --- remember, the rate at
which this happens is proportional to $1/N$. However, now imagine that a
ruthless male despot comes along and forbids all other males from mating. While
there are still $N=1,000$ individuals on this island, within a generation
*everyone* is the despot's kin.  The rate at which everyone on the island
becomes a cousin is now much faster.  Although this situation seems quite
different to the free love island, it only requires adjusting $N$;
specifically, we use an effective population size $N_e$. In this case, it works
out that the adjusted $N_e \approx 8$<sup></sup>.

Importantly, effective population also depends on population bottlenecks. If
instead of the male despot, free love island goes 99 generations undisturbed,
but then has a severe generation-long bottleneck where only 2 people survive,
the effective population size is now reduced from $N=1,000$ to $N_e = 170$.
Since it's effective population sizes, and not census sizes, that determine
levels of genetic variability, one can see how frequent population bottlenecks
could explain Lewontin's Paradox. The evolutionary history of many species also
includes range expansions and colonization and extinctions --- when a few
individuals make it a new patch of habitable land, procreate, but ultimately
may die. These dynamics also decrease effective population size and capture the
messy reality of many species.

<figure>
<img src="/images/hewitt_iceage.png" width="460" class="img-responsive"/>
  <figcaption>

  The earth looked very different over the last 20,000 years than it did today.
  Numerous ice sheets formed and forced species into refugia, where they may
  have diverged into different species or subspecies (for example, the <a
  href="https://en.wikipedia.org/wiki/Hooded_crow">hooded</a> and <a
  href="https://en.wikipedia.org/wiki/Carrion_crow">carrion</a> crows. This
  figure is from <a href="https://www.nature.com/articles/35016000">Hewitt
  (2000)</a>, <em>The genetic legacy of the Quaternary ice ages</em>.

  </figcaption>
</figure>

Kimura argued that the only "serious problem that remains to be explained" with
Lewontin's Paradox was that observed heterozygosities never exceeded 30%.
Considering *Drosophila* species, which were at the upper end of the population
size and heterozygosity ranges, Kimura argued that their numbers were severely
depressed by the last continental glaciation. Genetic variability takes a while
to recover from a severe bottleneck, and longer for larger populations (as new
mutations must escape rarity and work up to intermediate frequencies), so this
severe, lasting bottlenecks during glaciations is the primary neutralist
hypothesis for Lewontin's Paradox.

## Recent Work on Lewontin's Paradox 

Technological improvements in genome sequencing revolutionized population
genomics, and have provided evidence that diversity within species is shaped by
a mix of past demographic events, such as bottlenecks, as well as by natural
selection. Lewontin's Paradox however remained unresolved through the genomics
era. There are a few nice pieces of recent work that set up some context for my
recent paper. The first of which was [Ellen Leffler et al.
(2012)](https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1001388),
which compiled estimates of diversity from genomic data. This was one of the
first papers I read as I grew interested in population genomics in 2013, and
certainly inspired me to continue in the field. In this paper, Leffler and
colleagues survey diversity for 167 species, and find an 800-fold difference
between the highest diversity species (the sea squirt *Ciona savignyi*) to the
lowest diversity species (the wild cat, *Lynx lynx*). Overall, they point out
the narrow range of diversity is still an open and important mystery.

The second paper is [J. Romiguier et al.
(2014)](https://www.nature.com/articles/nature13685), which surveyed diversity
and life-history characteristics across 72 species. These authors find that
diversity levels across species are highly-correlated to ecological strategy,
in particular, whether the species have lots of offspring they invest little
in, or few offspring they invest more in. Intriguingly, these results suggest
ecological processes are predictive of genetic diversity across species.  While
these ecological correlates are an important piece of the puzzle, it's still
uncertain how mechanistically such ecological processes would act to constrain
genetic diversity.

More recently, [Russ Corbett-Detig, et al.
(2015)](https://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1002112)
tested the hypothesis of whether linked selection could explain Lewontin's
Paradox. Since population sizes are very difficult to estimate for many
species, Russ and colleagues considered how the strength of linked selection
varies with two proxies of population size: range size and body size.
Large-bodied animals (e.g. whales) have smaller population densities than
small-bodied animals (e.g. mice), in part because of the energy requirements
are much higher needed to sustain life in larger-bodied animals. Similarly,
we'd expect species with larger ranges to have larger population sizes than
species with small ranges. Then, the authors fit a statistical model to
estimate the strength of linked selection across population genomic samples for
40 species. As a former bioinformatician, I should point out this is a
monumental task worthy of praise. Interestingly, they find that the strength of
linked selection does seem to scale with these proxies of population size. I
read this paper in my first year of graduate school and discussions about it
with my PhD advisor Graham Coop were foundational to my early interest in
Lewontin's Paradox.

Graham ultimately found a limitation of Corbett-Detig et al. (2015), which he
shared on BioRxiv ([Coop
2016](https://www.biorxiv.org/content/10.1101/042598v1)). While Corbett-Detig
et al. do indeed find that linked selection is *stronger* in species with large
population sizes, is still may not be strong *enough* to explain why the range
of genetic diversity is so narrow. Ultimately, it seemed that the magnitude of
the impact of linked selection was reducing diversity some 60%-80% in species
with large population sizes, this isn't enough to explain why there's only an
800-fold difference between fruit flies and Lynx, when is likely well upwards
of a *million*-fold difference in their population sizes. The mystery remained
open, and at the end of Graham's article it seemed a bit more likely that only
repeated bottlenecks and other non-selective processes could explain the narrow
range of genetic diversity.

## Estimating Census Population Sizes

This brings us to my recent paper in eLife that hopefully adds a few more
pieces to this longstanding puzzle. The first component was to try to actually
estimate the population sizes across various animal species, and quantify the
relationship between genetic diversity and population size. Early work from
Soulé (1976), Frankham (1996), and Nei and Graur (1984) has looked at this
relationship before, using early protein heterozygosity data:

<figure>
<img src="/images/frankham_soule_nei_graur.png" width="660" class="img-responsive"/>
  <figcaption>
  (A) Frankham (1996) using the data of Soulé (1976), and (B) Nei and Graur
  (1984). <br/>The solid line is neutral expectation if the mutation rate is $\mu =
  10^{-7}$.
  </figcaption>
</figure>

These papers used population size estimates from the literature, or
back-of-the-napkin calculations. To get a modern look at the relationship
between genomic estimates of genetic diversity from previous studies (Leffler
et al. 2012, Romiguier et al. 2014, and Corbett-Detig et al. 2015) wanted to
find a way to roughly approximate census population sizes for hundreds of
species in an automated way. One approach is to take the product of population
density (i.e. how many individuals there are per square kilometer) and species
range size (i.e. how wide a species range is), to get a very crude estimate of
population size. The downside of this approach though, is that population
densities are unknown for most species too.

Luckily, the field of macroecology provides a way out. Animals have energetic
needs that scale with their body sizes. Large-bodied animals require more
energy that they procure by hunting or grazing a certain area; competition
between individuals for such resources means that large-bodied animals can only
live at lower densities. The shocking result is that population densities are
surprisingly well correlated with body mass. Damuth (1981, 1987) quantified
this; here is a figure from my paper (Figure 1--figure supplement 1):

<figure>
<img src="/images/damuth.png" width="560" class="img-responsive"/>
  <figcaption>
  The relationship between population density and population body mass from Damuth's 
  1987 data.
  </figcaption>
</figure>

Using this relationship, I predict population density using body mass. In
practice, I use body length, since this is much easier to collect and reported
in a more standard way than body mass. Using a statistical routine, I predict
body mass from body length, and population density from body mass.

Next, I need to estimate range size. A common approach is to use occurrence
data ---records of the latitude and longitude of where animals were observed---
to infer the range. Using the [Global Biodiversity Information
Facility](https://www.gbif.org/) database, I downloaded occurrence data for the
animals species I had genetic diversity and population density estimates for. I
then wrote some R code to automatically infer the ranges from this occurrence
data. 

<figure>
<img src="/images/apis_mellifera.png" width="660" class="img-responsive"/>
  <figcaption>
  The GBIF occurrence data (red points) and inferred range (green polygons) of
  the common honeybee (*Apis mellifera*). This shows some of the limitation of
  this course-grained approach: there is some uncertainty whether the some
  regions have sparse observations and the algorithm is properly filling in the
  range, or whether these are aberrant observations in regions where honeybees
  don't normally live. Still, for Lewontin's Paradox, we do not need accurate
  estimates, but rather a rough look. Compare the range of the honeybee [to
  that of](images/apis_cerana.png) *Apis cerana*, which lives only in South
  Asia.
  </figcaption>
</figure>

With the population densities and ranges estimated, I take their product to get
an approximate population size (see [Figure
1](https://elifesciences.org/articles/67509#fig1) of the paper for a look at
the distribution of these ranges by phylum).  There's quite a bit of validation
I do to ensure the numerous approximations I've made here are reasonable. For
example, I make sure my census sizes don't lead to predictions of the total
biomass that are unreasonably small or large (see [*Population Size
Validation*](https://elifesciences.org/articles/67509#s4) and [*Table
1*](https://elifesciences.org/articles/67509#table1)), since estimates of the
total biomass on earth of certain animal groups are available thanks to the
study of [Bar-On et al. (2018)](https://www.pnas.org/content/115/25/6506). I
also do lots of other consistency checks in [*Appendix
3*](https://elifesciences.org/articles/67509#appendix-3). I am especially
thankful to reviewer Guy Sella, who suspected my early fruitfly population
sizes were a tad too large; indeed, after lots of sleuthing, I found that
macroecologists like Damuth (1987) apply a correction to animals that do not
regulate their body temperature that I did not apply in earlier versions.

## The Relationship Between Genetic Diversity and Population Size

Next, I merged the approximate population sizes with the genetic diversity data
from Leffler et al. (2012), Romiguier et al. (2014), and Corbett-Detig et al.
(2015). These data allow for a nice visualization of Lewontin's Paradox of
Variation, through the relationship between genetic diversity and population
size:

<figure>
<img src="/images/diversity_popsize_full.png" width="860" class="img-responsive"/>
  <figcaption>

  The relationship between genetic diversity ($\pi$) and approximate population
  size ($N_c$) for 172 animal species. Note that genetic diversity varies just
  over three orders of magnitude, while census sizes vary over 12 orders of
  magnitude. The gray ribbon indicate the expected neutral diversity for a
  range of mutation rates ($10^{-9} \lt \mu \lt 10^{-8}$) were the diversity to
  be determined entirely by census size under the neutral model. Lewontin's
  Paradox wouldn't be a paradox if the diversity estimates fell in this thin
  gray area; instead they do not scale with population size, and are mostly
  constrained to within three orders of magnitude. The eLife article has
  numerous supplementary figures relevant to this figure
  [here](https://elifesciences.org/articles/67509#fig2).

</figcaption>
</figure>

What this relationship tells us is that population size appears to impact
genetic diversity in the way we'd expected (there is higher genetic diversity
in species with larger population sizes -- this is shown by the dashed gray
line of best fit in the figure above), but as Lewontin first pointed out,
genetic diversity doesn't increase as fast as we'd expect if solely census
population sizes and neutral evolution determined genetic diversity.

## Is the Diversity--Population-Size relationship meaningful?

<figure>
<img src="/images/darwin_i_think.png" width="560" class="img-responsive"/>
  <figcaption>
  Darwin's sketch of a phylogenetic tree in his *First Notebook on
  Transmutation of Species*, 1837. Source: [Wikimedia
  commons](https://commons.wikimedia.org/wiki/File:Darwin_Tree_1837.png).
</figcaption>
</figure>

We might look at the relationship above and conclude that there is a
significant relationship between population sizes and genetic diversity.
Unfortunately, there's a statistical conundrum that crops up whenever we draw
lines of best fit through data points like species. Species are connected by an
underlying *phylogenetic tree*; certain species share common ancestors more
recently than others. This was one of Darwin's brilliant ideas, sketched out in
1837 in his notebook (figure above). In a seminal article in 1985, Joe
Felsenstein pointed out that the underlying species tree creates a statistical
problem whenever we want to make comparisons across species (as I'm doing
here), and this statistical issue should be accounted for.

We can understand this statistical issue, known as phylogenetic
non-independence, with a story. Imagine going to a large family reunion, where
there are two sides of the family: all the descendents of your
great-great-grandfather, and all the descendents of his sister, your
great-great-aunt. You look around at all your family members, and observe that
there seems to be a statistical relationship between having freckles and being
tall. Your tallest relatives tend to have the most freckles, including your
great-great aunt's husband, who towers over you. Other relatives are more
average height, and have less freckles. If you were to collect data and
quantify this relationship, you may conclude that somehow these two traits are
correlated in a meaningful way; perhaps there's some biological process that
underpins both characteristics.

However, this ignores something: all the relatives that are tall and have
freckles descend from your great-great aunt and uncle, and all the average
height relatives without freckles descend from your great-great grandfather
(who is shorter and doesn't have freckles). There isn't necessarily a
meaningful relationship between these two traits --- it could just be an
accident of who shares ancestry (and thus the genes that determine traits) with
whom. The family tree here breaks the statistical assumption of *independence*,
since everyone's related and thus not independent of one another.

This same problem occurs when we compare traits across species too, traits like
genetic diversity and census size. John Gillespie, who I introduced at the
start of this article, suspected that early pictures of the
diversity--population-size relationship may be *entirely* a spurious artifact
of this shared ancestry. In his 1991 book, he has the following figure:

<figure>
<img src="/images/gillespie.png" width="460" class="img-responsive"/>
  <figcaption>
  John Gillespie suspected that the relationship between population size and
  genetic diversity found by Nei and Graur (1984) could be entirely an artifact
  of two groups of animals: carnivores and fruitflies (1991).
  </figcaption>
</figure>

While some previous work has addressed this statistical quandary in interesting
ways, it's been limited by the difficulty of building big species phylogenetic
trees. In my paper, I use a tool called
[datelife](https://github.com/phylotastic/datelife) to build these trees, and
then account for this tree structure using a statistical model known as
Bayesian mixed-effect models. 

<figure>
<img src="/images/diversity_pcm.png" width="860" class="img-responsive"/>
  <figcaption>
  </figcaption>
</figure>


<!--
Population geneticists have a variety of ways to measure genetic
diversity from these gene frequencies, but perhaps the most common is 

and what seemed to be so strange at the time that people felt they had to take
sides. I remember it was at it's height when Dick Lewontin spent a year in my
lab here, and I remember we talked about it, and we both agree at the moment we
couldn't see anything really decisive one way or the other. And we kept on
trying to think of decisive statistical measures or experiments one could carry
out to decide whether the neutral theory was true, and we kept coming up with
ideas and they deciding wouldn't really settle the issue because... you know...
and so we decided the only sensible thing to do was to say we don't know it
didn't seem necessary to say in science you know when you don't, but almost
everyone else seemed to take rigorous sides sides.


a great evolutionary geneticist named
Richard Lewontin measured the genetic variability within proteins using an
older technology. In his seminal paper with Jack Hubby (Hubby and Lewontin,
    1966) that provided the first ever measure of genetic variability<sup>2</sup>
(in the fruitfly *Drosophila pseudoobscura*), they write:




Population geneticists have a variety of ways to
measure genetic diversity, but perhaps the most basic is to just count the
differences between two DNA sequences like `AGCCAGATAC` and `AGCGAGTTAC` in
this case there are two differences, a `G` in the fourth position and a `T` in
the seventh position in the second sequence.


Whenever I go on a run along a small creek here in Eugene, I try to notice as
many species of flora and fauna as I can: Stellar's jay, western tiger
swallowtail, turkey vulture, sitka willow, garter snake, western swordfern.
These species have varied evolutionary histories on this planet. The western
tiger swallowtail occupies most states west of Texas, but its genus, *Papilio*,
      is some [55-65 million years
      old](https://academic.oup.com/sysbio/article/53/2/278/1687251) and is found
      worldwide. By contrast, the northwestern garter snakes (*Thamnophis
          ordinoides*) that slither into the grass to dodge runners only reside in the
      Pacific Northwest, and members of their genus are only found in North America.

      If I were to go out on my next run and collect blood samples from different
      individuals for all the species I encountered, then extract and sequence their
      DNA, I could measure the **genetic diversity** within each species. Genetic
      diversity is measure of how much genetic variability there is in a population,
      and for some reason we denote this quantity with the Greek letter $\pi$.
      Imagine counting the differences between the As, Ts, Cs, and Gs of two
      chromosomes from two random humans on earth- that would be an estimate
      genetic diversity of humans, it's approximately $\pi_\text{humans} \approx
      0.001$ since roughly 1 in every thousand bases differ between any two random
      human chromosomes.

## Lewontin and Hubby's Quest to Measure Genetic Variability

      Long before we had DNA sequencers, a great evolutionary geneticist named
      Richard Lewontin measured the genetic variability within proteins using an
      older technology. In his seminal paper with Jack Hubby (Hubby and Lewontin,
          1966) that provided the first ever measure of genetic variability<sup>2</sup>
      (in the fruitfly *Drosophila pseudoobscura*), they write:

      > In a sense, a description of the genetic variation in a population is the
      > fundamental datum of evolutionary studies; and it is necessary to explain the
      > origin and maintenance of this variation and to predict its evolutionary
      > consequences. [...]

      At the time, evolutionary biologists lacked a sense of how high or low the
      genetic variability was within populations. They had some sense of the
      frequencies of visible polymorphisms (such as the banding polymorphisms of
          *Cepaea nemoralis* snails or the wing color variants in scarlet tiger moths),
      and large chromosomal rearrangements that could be seen with a microscope, like
      inversions. However, these types of polymorphisms aren't a fair look at the
      central quantity of evolution that Lewontin wanted to measure: how different
      are two random chromosomes (or, using the terminology of genetics, what
          proportion of loci are heterozygous?). Trying to get a sense of levels of
      *genetic polymorphism* from observable *visible polymorphisms* could be biased:
      you may be looking through the same lens a predator does when it exerts
      selection by eating the variants that it can see, or those that look most
      appetizing. 

      At the time, there were two opposed views of evolution. The **classic theory**
      supposed that most individuals were very close to a "wild-type"<sup>3</sup>,
      and deviations from this wild-type were almost entirely due to harmful rare
      mutations that were kept at low frequency by selection. The alternative view,
      which emerged from Dobzhanksy's study of how inversion polymorphisms in
      *Drosophila pseudoobscura* varied across space and through the seasons,
      believed that genetic variants were at mostly intermediate frequencies either
      because heterozygous<sup>4</sup> individuals were the most fit (overdominance),
      or some genetic variants became more fit when they became more rare
      (frequency-dependent selection). These two diametrically opposed theories were
      at loggerheads for decades in part because we didn't have the data to answer a
      simple question: how much genetic variability was there in a species?

      It's an unnamed phenomenon in science that theoretic disagreements become
      entrenched idealogical battles when there is insufficient data to kill one
      theory in favor of the other. Hubby and Lewontin's study of genetic variation
      in *Drosophila pseudoobscura* (as well as Harri's 1966 study of protein
          variability in humans) was the data that killed the original classic theory for
      a simple reason: Lewontin and Hubby discovered *Drosophila pseudoobscura*
      harbored way too genetic variability within proteins. This was wholly
      inconsistent with the idea of populations consisting of a predominant wild-type
      and a few odd rare deleterious mutations on their way to
      extinction.<sup>4</sup>

      <aside>
      Yet, the balance hypothesis was also in grave danger: while there was indeed a
      lot of variability within populations, experiments inbreeding *Drosophila* 
      </aside>



      set forth nothing short of a empirical revolution in
      population genetics. Soon Nevo, Clegg, Allard 






      Manifold consequences flow from the assumption of of the classical or balance
      hypothesis. If the classic hypothesis were correct, the difference between
      populations would be of far more profound significance than under the balance
      theory. Since there would be so little genetic variability between individuals
      within populations, most of the genetic diversity within a polymorphic species
      would be inter-populational. In man, the manifest genetic differences between
      geographical races would represent a much greater proportion of total human
      variation than occurs within races, giving race a considerable biological
      importance. A basis for racism may also flow from the concept of a wild type,
      since if there is genetic type of the species, those who fail to correspond to
      it must be less perfect. Platonic notions of type are likely to intrude
      themselves from one domain to another, and Dobzhanksy (1955) was clearly
      conscious of this problem when he attacked the concept of a wild type.  

      -->
